stacks-signer/src/signerdb.rs

```diff
@@ -70,6 +81,41 @@ impl SignerDb {
         )
     }
 
+    /// Get the signer state for the provided reward cycle if it exists in the database
+    pub fn get_signer_state(&self, reward_cycle: u64) -> Result<Option<SignerState>, DBError> {
+        let result: Option<String> = query_row(
+            &self.db,
+            "SELECT state FROM signer_states WHERE reward_cycle = ?",
+            &[reward_cycle.to_string()],
```

Comment:
I think it's bad to `to_string()` a u64 to an integer. This leads to sqlite's conversion behavior, and I don't really know what it would do in the case of trying to pass `(i64::MAX as u64) + 1` here. We use `u64_to_sql()` for this.

Reply:
Done :heavy_check_mark: